src/sirocco/parsing/yaml_data_models.py

```diff
@@ -273,58 +274,22 @@ class ConfigRootTask(ConfigBaseTask):
-@dataclass(frozen=True)
-class ShellCliArgument:
-    """A holder for a CLI argument to simplify access.
-
-    Stores CLI arguments of the form "file", "--init", "{file}" or "{--init file}". These examples translate into
-    ShellCliArguments ShellCliArgument(name="file", references_data_item=False, cli_option_of_data_item=None),
-    ShellCliArgument(name="--init", references_data_item=False, cli_option_of_data_item=None),
-    ShellCliArgument(name="file", references_data_item=True, cli_option_of_data_item=None),
-    ShellCliArgument(name="file", references_data_item=True, cli_option_of_data_item="--init")
-
-    Attributes:
-        name: Name of the argument. For the examples it is "file", "--init", "file" and "file"
-        references_data_item: Specifies if the argument references a data item signified by enclosing it by curly
-            brackets.
-        cli_option_of_data_item: The CLI option associated to the data item.
-    """
-
-    name: str
-    references_data_item: bool
-    cli_option_of_data_item: str | None = None
-
-    def __post_init__(self):
-        if self.cli_option_of_data_item is not None and not self.references_data_item:
-            msg = "data_item_option cannot be not None if cli_option_of_data_item is False"
-            raise ValueError(msg)
-
-    @classmethod
-    def from_cli_argument(cls, arg: str) -> ShellCliArgument:
-        len_arg_with_option = 2
-        len_arg_no_option = 1
-        references_data_item = arg.startswith("{") and arg.endswith("}")
-        # remove curly brackets "{--init file}" -> "--init file"
-        arg_unwrapped = arg[1:-1] if arg.startswith("{") and arg.endswith("}") else arg
-
-        # "--init file" -> ["--init", "file"]
-        input_arg = arg_unwrapped.split()
-        if len(input_arg) != len_arg_with_option and len(input_arg) != len_arg_no_option:
-            msg = f"Expected argument of format {{data}} or {{option data}} but found {arg}"
-            raise ValueError(msg)
-        name = input_arg[0] if len(input_arg) == len_arg_no_option else input_arg[1]
-        cli_option_of_data_item = input_arg[0] if len(input_arg) == len_arg_with_option else None
-        return cls(name, references_data_item, cli_option_of_data_item)
-
-
 @dataclass(kw_only=True)
 class ConfigShellTaskSpecs:
     plugin: ClassVar[Literal["shell"]] = "shell"
-    command: str = ""
-    cli_arguments: list[ShellCliArgument] = field(default_factory=list)
-    env_source_files: list[str] = field(default_factory=list)
+    port_pattern: ClassVar[re.Pattern] = field(default=re.compile(r"{PORT::(.*?)}"), repr=False)
     src: str | None = None
+    command: str
+    multi_arg_sep: str = " "
+    env_source_files: list[str] = field(default_factory=list)
+
+    def resolve_ports(self, input_labels: dict[str, list[str]]) -> str:
+        """returns a string corresponding to self.command with {PORT::...}
+        placeholders replaced by the content provided in the input_labels dict"""
```

Comment:
```suggestion
        placeholders replaced by the content provided in the input_labels dict. To support arguments like `tool input-date1 input-date2` using whitespaces as separator, or `tool --files=input-date1,input-date2` or using comma as separator, or `tool --option date1 --option date2` using "--option" as separator.
        """
```
Could you also add docstring test for this function? It takes some time to fully understand it.

Reply:
I do agree